pkg/component/crddeployer/crd.go

```diff
@@ -62,6 +65,13 @@ func (c *crdDeployer) Destroy(ctx context.Context) error {
 					Name: resourceName,
 				},
 			}
+
+			if c.confirmDeletion {
+				if err := gardenerutils.ConfirmDeletion(ctx, c.client, crd); client.IgnoreNotFound(err) != nil {
```

Comment:
You should get the crd first instead of passing a nearly empty object.

See https://github.com/timuthy/gardener/commit/ce76026cf865bc3dfdcb2df1ab037617639b60b0 for reference.